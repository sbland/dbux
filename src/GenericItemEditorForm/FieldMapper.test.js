import { mapFields } from './FieldMapper';

const productParameters = [
  {
    uid: 'product-code',
    label: 'Product Code',
    type: 'text',
    required: true,
    group: 0,
  },
  {
    uid: 'product-standard',
    label: 'Product Standard',
    type: 'select',
    options: [
      {
        uid: 'bsen',
        label: 'BSEN',
      },
      {
        uid: 'ansi',
        label: 'ANSI',
      },
    ],
    required: true,
    group: 0,
  },
  {
    uid: 'finish',
    label: 'Finish',
    type: 'reference',
    collection: 'productfinishes',
    group: 1,
    objectLink: true,
  },

  {
    uid: 'fire-rating',
    label: 'Fire Rating',
    type: 'select',
    influencer: true,
    group: 1,
    options: [
      {
        uid: 'FR30',
        label: 'FR30',
      },
      {
        uid: 'FR45',
        label: 'FR45',
      },
      {
        uid: 'FR60',
        label: 'FR60',
      },
    ],
  },
  {
    uid: 'cost',
    label: 'Cost',
    type: 'number',
    group: 2,
    required: true,
  },

  {
    uid: 'images',
    label: 'Images',
    multiple: true,
    fileType: 'image',
    type: 'image',
    group: 3,
  },
  {
    uid: 'quantity',
    label: 'Quantity',
    type: 'number',
    readOnly: true,
    group: 4,
  },
];
describe('Item Editor Field Mapper', () => {
  test('should map fields to headings', () => {
    // TODO: Finish implementing this test
    const params = productParameters;
    const overridenFields = [];
    const productId = 'prodId';
    const collection = 'products';
    const headings = mapFields(params, overridenFields, productId, collection);
    expect(headings).toMatchSnapshot();
  });
  test('should make quantity editable if is set to editable', () => {
    const quantityParam = {
      uid: 'quantity',
      label: 'Quantity',
      type: 'number',
      readOnly: true,
      group: 1,
    };
    const params = [quantityParam];
    const overridenFields = [];
    const productId = 'prodId';
    const headings = mapFields(params, overridenFields, productId, 'products');
    expect(headings[0].children).toEqual([{ ...quantityParam, readOnly: true }]);
  });
});
