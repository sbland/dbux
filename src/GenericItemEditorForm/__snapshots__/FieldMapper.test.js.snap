// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Editor Field Mapper should map fields to headings 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "group": 0,
        "label": "Product Code",
        "required": true,
        "type": "text",
        "uid": "product-code",
      },
      Object {
        "group": 0,
        "label": "Product Standard",
        "options": Array [
          Object {
            "label": "BSEN",
            "uid": "bsen",
          },
          Object {
            "label": "ANSI",
            "uid": "ansi",
          },
        ],
        "required": true,
        "type": "select",
        "uid": "product-standard",
      },
    ],
    "label": "",
    "orientation": "horiz",
    "type": "embedded",
    "uid": "group-0",
  },
  Object {
    "children": Array [
      Object {
        "collection": "productfinishes",
        "group": 1,
        "label": "Finish",
        "objectLink": true,
        "type": "reference",
        "uid": "finish",
      },
      Object {
        "group": 1,
        "influencer": true,
        "label": "Fire Rating",
        "options": Array [
          Object {
            "label": "FR30",
            "uid": "FR30",
          },
          Object {
            "label": "FR45",
            "uid": "FR45",
          },
          Object {
            "label": "FR60",
            "uid": "FR60",
          },
        ],
        "type": "select",
        "uid": "fire-rating",
      },
    ],
    "label": "",
    "orientation": "horiz",
    "type": "embedded",
    "uid": "group-1",
  },
  Object {
    "children": Array [
      Object {
        "group": 2,
        "label": "Cost",
        "required": true,
        "type": "number",
        "uid": "cost",
      },
    ],
    "label": "",
    "orientation": "horiz",
    "type": "embedded",
    "uid": "group-2",
  },
  Object {
    "children": Array [
      Object {
        "collectionId": "products",
        "documentId": "prodId",
        "fileType": "image",
        "group": 3,
        "label": "Images",
        "multiple": true,
        "type": "image",
        "uid": "images",
      },
    ],
    "label": "",
    "orientation": "horiz",
    "type": "embedded",
    "uid": "group-3",
  },
  Object {
    "children": Array [
      Object {
        "group": 4,
        "label": "Quantity",
        "readOnly": true,
        "type": "number",
        "uid": "quantity",
      },
    ],
    "label": "",
    "orientation": "horiz",
    "type": "embedded",
    "uid": "group-4",
  },
]
`;
